### ISSUE TITLE
Gemini calls not work

### ISSUE THREAD

**Issue by hernanjls** [chataize/generative-cs#1]
System.Net.Http.HttpRequestException: 'Response status code does not indicate success: 400 (Bad Request).'

using ChatAIze.GenerativeCS.Clients;
using ChatAIze.GenerativeCS.Models;
using ChatAIze.GenerativeCS.Options.Gemini;

var client = new GeminiClient("gtxxxxxxxxaOwsDlP7YmPzrGoU");
var chat = new Chat();

var options = new ChatCompletionOptions
{
    IsTimeAware = true,
    IsDebugMode = true
};

options.AddFunction("Check City Temperature", (string city) => {
    Random.Shared.Next(0, 100);
   });
options.AddFunction("Check Country Temperature", async (string country) => {
    await Task.Delay(5000);
    return new { Temp = Random.Shared.Next(0, 100) };
});

while (true)
{
    var input = Console.ReadLine();
    if (string.IsNullOrWhiteSpace(input))
    {
        break;
    }

    chat.FromUser(input);

    var response = await client.CompleteAsync(chat, options);
    Console.WriteLine(response);
}

**Comment by marcel2215** [comment]
Thank you for reporting the issue! The problem is that `additionalProperties` is exclusive to Open AI client.

```json
{
    "error": {
        "code": 400,
        "message": "Invalid JSON payload received. Unknown name \"additionalProperties\" at 'tools[0].function_declarations[0].parameters': Cannot find field.",
        "status": "INVALID_ARGUMENT",
        "details": [
            {
                "@type": "type.googleapis.com/google.rpc.BadRequest",
                "fieldViolations": [
                    {
                        "field": "tools[0].function_declarations[0].parameters",
                        "description": "Invalid JSON payload received. Unknown name \"additionalProperties\" at 'tools[0].function_declarations[0].parameters': Cannot find field."
                    }
                ]
            }
        ]
    }
}
```

i will fix it in the next update.

**Comment by marcel2215** [comment]
Fixed in Generative CS **0.11.20**: https://github.com/chataize/generative-cs/releases/tag/v0.11.20
